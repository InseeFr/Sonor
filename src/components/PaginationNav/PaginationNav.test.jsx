// Link.react.test.js
import React from 'react';
import {
  render, screen, cleanup,
} from '@testing-library/react';
import PaginationNav from './PaginationNav';

afterEach(cleanup);

const pagination = { size: 5, page: 1 };

const updateFunc = jest.fn();

it('Page selector display: 2 items / size = 5', async () => {
  const component = render(
    <PaginationNav.PageSelector
      pagination={pagination}
      updateFunc={updateFunc}
      numberOfItems={2}
    />,
  );
  // Should match snapshot (only one page)
  expect(component).toMatchSnapshot();
  expect(component.baseElement.querySelectorAll('.page-item').length).toEqual(1);
});

it('Page selector display: 8 items / size = 5', async () => {
  const component = render(
    <PaginationNav.PageSelector
      pagination={pagination}
      updateFunc={updateFunc}
      numberOfItems={8}
    />,
  );
  // Should match snapshot (2 page)
  expect(component.baseElement.querySelectorAll('.page-item').length).toEqual(2);
  expect(component).toMatchSnapshot();
});

it('Page selector display: 16 items / size = 5', async () => {
  const component = render(
    <PaginationNav.PageSelector
      pagination={pagination}
      updateFunc={updateFunc}
      numberOfItems={16}
    />,
  );
  // Should match snapshot (4 pages)
  expect(component.baseElement.querySelectorAll('.page-item').length).toEqual(4);
  expect(component).toMatchSnapshot();
});
it('Page selector display: 22 items / size = 5', async () => {
  const component = render(
    <PaginationNav.PageSelector
      pagination={pagination}
      updateFunc={updateFunc}
      numberOfItems={22}
    />,
  );
  // Should match snapshot (5 pages)
  expect(component).toMatchSnapshot();
});

it('Page selector display: 28 items / size = 5', async () => {
  const component = render(
    <PaginationNav.PageSelector
      pagination={pagination}
      updateFunc={updateFunc}
      numberOfItems={28}
    />,
  );
  // Should match snapshot (6 pages)
  expect(component.baseElement.querySelectorAll('.page-item').length).toEqual(6);
  expect(component).toMatchSnapshot();
});

it('Page selector display: 105 items / size = 5', async () => {
  const component = render(
    <PaginationNav.PageSelector
      pagination={pagination}
      updateFunc={updateFunc}
      numberOfItems={105}
    />,
  );
  // Should match snapshot (fist page, second, third, "...", last)
  expect(component.baseElement.querySelectorAll('.page-item').length).toEqual(7);
  expect(component).toMatchSnapshot();
});

it('Page selector display: 105 items / size = 5 on page 10', async () => {
  const component = render(
    <PaginationNav.PageSelector
      pagination={{ page: 10, size: 5 }}
      updateFunc={updateFunc}
      numberOfItems={105}
    />,
  );

  // Should match snapshot (fist page, "...", 9, 10, 11, "...", last)
  expect(component.baseElement.querySelectorAll('.page-item').length).toEqual(7);
  expect(component).toMatchSnapshot();
});

it('Page selector display: 105 items / size = 5 on page 20', async () => {
  const component = render(
    <PaginationNav.PageSelector
      pagination={{ page: 20, size: 5 }}
      updateFunc={updateFunc}
      numberOfItems={105}
    />,
  );

  // Should match snapshot (fist page, "...", 16, 17, 18, 19, 20)
  expect(component.baseElement.querySelectorAll('.page-item').length).toEqual(7);
  expect(component).toMatchSnapshot();
});
