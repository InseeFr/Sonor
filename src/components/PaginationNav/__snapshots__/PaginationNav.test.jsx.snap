// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page selector display: 2 items / size = 5 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="paginationNav pagination pagination-sm"
        data-testid="pagination-nav"
      >
        <li
          class="page-item active"
        >
          <span
            class="page-link"
          >
            1
            <span
              class="sr-only"
            >
              (current)
            </span>
          </span>
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="paginationNav pagination pagination-sm"
      data-testid="pagination-nav"
    >
      <li
        class="page-item active"
      >
        <span
          class="page-link"
        >
          1
          <span
            class="sr-only"
          >
            (current)
          </span>
        </span>
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Page selector display: 8 items / size = 5 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="paginationNav pagination pagination-sm"
        data-testid="pagination-nav"
      >
        <li
          class="page-item active"
        >
          <span
            class="page-link"
          >
            1
            <span
              class="sr-only"
            >
              (current)
            </span>
          </span>
        </li>
        <li
          class="page-item"
        >
          <a
            class="page-link"
            href="#"
            role="button"
          >
            2
          </a>
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="paginationNav pagination pagination-sm"
      data-testid="pagination-nav"
    >
      <li
        class="page-item active"
      >
        <span
          class="page-link"
        >
          1
          <span
            class="sr-only"
          >
            (current)
          </span>
        </span>
      </li>
      <li
        class="page-item"
      >
        <a
          class="page-link"
          href="#"
          role="button"
        >
          2
        </a>
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Page selector display: 16 items / size = 5 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="paginationNav pagination pagination-sm"
        data-testid="pagination-nav"
      >
        <li
          class="page-item active"
        >
          <span
            class="page-link"
          >
            1
            <span
              class="sr-only"
            >
              (current)
            </span>
          </span>
        </li>
        <li
          class="page-item"
        >
          <a
            class="page-link"
            href="#"
            role="button"
          >
            2
          </a>
        </li>
        <li
          class="page-item"
        >
          <a
            class="page-link"
            href="#"
            role="button"
          >
            3
          </a>
        </li>
        <li
          class="page-item"
        >
          <a
            class="page-link"
            href="#"
            role="button"
          >
            4
          </a>
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="paginationNav pagination pagination-sm"
      data-testid="pagination-nav"
    >
      <li
        class="page-item active"
      >
        <span
          class="page-link"
        >
          1
          <span
            class="sr-only"
          >
            (current)
          </span>
        </span>
      </li>
      <li
        class="page-item"
      >
        <a
          class="page-link"
          href="#"
          role="button"
        >
          2
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          class="page-link"
          href="#"
          role="button"
        >
          3
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          class="page-link"
          href="#"
          role="button"
        >
          4
        </a>
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Page selector display: 22 items / size = 5 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="paginationNav pagination pagination-sm"
        data-testid="pagination-nav"
      >
        <li
          class="page-item active"
        >
          <span
            class="page-link"
          >
            1
            <span
              class="sr-only"
            >
              (current)
            </span>
          </span>
        </li>
        <li
          class="page-item"
        >
          <a
            class="page-link"
            href="#"
            role="button"
          >
            2
          </a>
        </li>
        <li
          class="page-item"
        >
          <a
            class="page-link"
            href="#"
            role="button"
          >
            3
          </a>
        </li>
        <li
          class="page-item"
        >
          <a
            class="page-link"
            href="#"
            role="button"
          >
            4
          </a>
        </li>
        <li
          class="page-item"
        >
          <a
            class="page-link"
            href="#"
            role="button"
          >
            5
          </a>
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="paginationNav pagination pagination-sm"
      data-testid="pagination-nav"
    >
      <li
        class="page-item active"
      >
        <span
          class="page-link"
        >
          1
          <span
            class="sr-only"
          >
            (current)
          </span>
        </span>
      </li>
      <li
        class="page-item"
      >
        <a
          class="page-link"
          href="#"
          role="button"
        >
          2
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          class="page-link"
          href="#"
          role="button"
        >
          3
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          class="page-link"
          href="#"
          role="button"
        >
          4
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          class="page-link"
          href="#"
          role="button"
        >
          5
        </a>
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Page selector display: 28 items / size = 5 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="paginationNav pagination pagination-sm"
        data-testid="pagination-nav"
      >
        <li
          class="page-item active"
        >
          <span
            class="page-link"
          >
            1
            <span
              class="sr-only"
            >
              (current)
            </span>
          </span>
        </li>
        <li
          class="page-item"
        >
          <a
            class="page-link"
            href="#"
            role="button"
          >
            2
          </a>
        </li>
        <li
          class="page-item"
        >
          <a
            class="page-link"
            href="#"
            role="button"
          >
            3
          </a>
        </li>
        <li
          class="page-item"
        >
          <a
            class="page-link"
            href="#"
            role="button"
          >
            4
          </a>
        </li>
        <li
          class="page-item"
        >
          <a
            class="page-link"
            href="#"
            role="button"
          >
            5
          </a>
        </li>
        <li
          class="page-item"
        >
          <a
            class="page-link"
            href="#"
            role="button"
          >
            6
          </a>
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="paginationNav pagination pagination-sm"
      data-testid="pagination-nav"
    >
      <li
        class="page-item active"
      >
        <span
          class="page-link"
        >
          1
          <span
            class="sr-only"
          >
            (current)
          </span>
        </span>
      </li>
      <li
        class="page-item"
      >
        <a
          class="page-link"
          href="#"
          role="button"
        >
          2
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          class="page-link"
          href="#"
          role="button"
        >
          3
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          class="page-link"
          href="#"
          role="button"
        >
          4
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          class="page-link"
          href="#"
          role="button"
        >
          5
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          class="page-link"
          href="#"
          role="button"
        >
          6
        </a>
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Page selector display: 105 items / size = 5 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="paginationNav pagination pagination-sm"
        data-testid="pagination-nav"
      >
        <li
          class="page-item active"
        >
          <span
            class="page-link"
          >
            1
            <span
              class="sr-only"
            >
              (current)
            </span>
          </span>
        </li>
        <li
          class="page-item"
        >
          <a
            class="page-link"
            href="#"
            role="button"
          >
            2
          </a>
        </li>
        <li
          class="page-item"
        >
          <a
            class="page-link"
            href="#"
            role="button"
          >
            3
          </a>
        </li>
        <li
          class="page-item"
        >
          <a
            class="page-link"
            href="#"
            role="button"
          >
            4
          </a>
        </li>
        <li
          class="page-item"
        >
          <a
            class="page-link"
            href="#"
            role="button"
          >
            5
          </a>
        </li>
        <li
          class="page-item"
        >
          <a
            class="page-link"
            href="#"
            role="button"
          >
            ...
          </a>
        </li>
        <li
          class="page-item"
        >
          <a
            class="page-link"
            href="#"
            role="button"
          >
            21
          </a>
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="paginationNav pagination pagination-sm"
      data-testid="pagination-nav"
    >
      <li
        class="page-item active"
      >
        <span
          class="page-link"
        >
          1
          <span
            class="sr-only"
          >
            (current)
          </span>
        </span>
      </li>
      <li
        class="page-item"
      >
        <a
          class="page-link"
          href="#"
          role="button"
        >
          2
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          class="page-link"
          href="#"
          role="button"
        >
          3
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          class="page-link"
          href="#"
          role="button"
        >
          4
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          class="page-link"
          href="#"
          role="button"
        >
          5
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          class="page-link"
          href="#"
          role="button"
        >
          ...
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          class="page-link"
          href="#"
          role="button"
        >
          21
        </a>
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Page selector display: 105 items / size = 5 on page 10 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="paginationNav pagination pagination-sm"
        data-testid="pagination-nav"
      >
        <li
          class="page-item"
        >
          <a
            class="page-link"
            href="#"
            role="button"
          >
            1
          </a>
        </li>
        <li
          class="page-item"
        >
          <a
            class="page-link"
            href="#"
            role="button"
          >
            ...
          </a>
        </li>
        <li
          class="page-item"
        >
          <a
            class="page-link"
            href="#"
            role="button"
          >
            9
          </a>
        </li>
        <li
          class="page-item active"
        >
          <span
            class="page-link"
          >
            10
            <span
              class="sr-only"
            >
              (current)
            </span>
          </span>
        </li>
        <li
          class="page-item"
        >
          <a
            class="page-link"
            href="#"
            role="button"
          >
            11
          </a>
        </li>
        <li
          class="page-item"
        >
          <a
            class="page-link"
            href="#"
            role="button"
          >
            ...
          </a>
        </li>
        <li
          class="page-item"
        >
          <a
            class="page-link"
            href="#"
            role="button"
          >
            21
          </a>
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="paginationNav pagination pagination-sm"
      data-testid="pagination-nav"
    >
      <li
        class="page-item"
      >
        <a
          class="page-link"
          href="#"
          role="button"
        >
          1
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          class="page-link"
          href="#"
          role="button"
        >
          ...
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          class="page-link"
          href="#"
          role="button"
        >
          9
        </a>
      </li>
      <li
        class="page-item active"
      >
        <span
          class="page-link"
        >
          10
          <span
            class="sr-only"
          >
            (current)
          </span>
        </span>
      </li>
      <li
        class="page-item"
      >
        <a
          class="page-link"
          href="#"
          role="button"
        >
          11
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          class="page-link"
          href="#"
          role="button"
        >
          ...
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          class="page-link"
          href="#"
          role="button"
        >
          21
        </a>
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Page selector display: 105 items / size = 5 on page 20 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="paginationNav pagination pagination-sm"
        data-testid="pagination-nav"
      >
        <li
          class="page-item"
        >
          <a
            class="page-link"
            href="#"
            role="button"
          >
            1
          </a>
        </li>
        <li
          class="page-item"
        >
          <a
            class="page-link"
            href="#"
            role="button"
          >
            ...
          </a>
        </li>
        <li
          class="page-item"
        >
          <a
            class="page-link"
            href="#"
            role="button"
          >
            17
          </a>
        </li>
        <li
          class="page-item"
        >
          <a
            class="page-link"
            href="#"
            role="button"
          >
            18
          </a>
        </li>
        <li
          class="page-item"
        >
          <a
            class="page-link"
            href="#"
            role="button"
          >
            19
          </a>
        </li>
        <li
          class="page-item active"
        >
          <span
            class="page-link"
          >
            20
            <span
              class="sr-only"
            >
              (current)
            </span>
          </span>
        </li>
        <li
          class="page-item"
        >
          <a
            class="page-link"
            href="#"
            role="button"
          >
            21
          </a>
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="paginationNav pagination pagination-sm"
      data-testid="pagination-nav"
    >
      <li
        class="page-item"
      >
        <a
          class="page-link"
          href="#"
          role="button"
        >
          1
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          class="page-link"
          href="#"
          role="button"
        >
          ...
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          class="page-link"
          href="#"
          role="button"
        >
          17
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          class="page-link"
          href="#"
          role="button"
        >
          18
        </a>
      </li>
      <li
        class="page-item"
      >
        <a
          class="page-link"
          href="#"
          role="button"
        >
          19
        </a>
      </li>
      <li
        class="page-item active"
      >
        <span
          class="page-link"
        >
          20
          <span
            class="sr-only"
          >
            (current)
          </span>
        </span>
      </li>
      <li
        class="page-item"
      >
        <a
          class="page-link"
          href="#"
          role="button"
        >
          21
        </a>
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
